auslib/db.py

```diff
@@ -796,22 +797,25 @@ def getRulesMatchingQuery(self, updateQuery, fallbackChannel, transaction=None):
                 self.log.debug(r)
         return matchingRules
 
-    def getRuleById(self, rule_id, transaction=None):
-        """ Returns the unique rule that matches the give rule_id """
-        rules = self.select(where=[self.rule_id == rule_id], transaction=transaction)
+    def getRule(self, id_or_alias, transaction=None):
+        """ Returns the unique rule that matches the give rule_id or alias."""
+        rules = self.select(
+            where=[(self.alias == id_or_alias) | (self.rule_id == id_or_alias)],
```

Comment:
Good point. I don't think it would be hugely slower to use the current query, but it should be easy and cheap to figure out what it is...so we may as well.